worlds/zork_grand_inquisitor/game_controller.py

```diff
@@ -0,0 +1,1447 @@
+                elif hotspot_item == ZorkGrandInquisitorItems.HOTSPOT_CANDY_MACHINE_BUTTONS:
+                    if self.game_state_manager.game_location == "tr5g":
+                        key: int
+                        for key in data.statemap_keys:
+                            self._write_game_flags_value_for(key, 0)
+                elif hotspot_item == ZorkGrandInquisitorItems.HOTSPOT_CANDY_MACHINE_COIN_SLOT:
+                    if self.game_state_manager.game_location == "tr5g":
+                        self._write_game_flags_value_for(12702, 0)
+                elif hotspot_item == ZorkGrandInquisitorItems.HOTSPOT_CANDY_MACHINE_VACUUM_SLOT:
+                    if self.game_state_manager.game_location == "tr5m":
+                        self._write_game_flags_value_for(12909, 0)
+                elif hotspot_item == ZorkGrandInquisitorItems.HOTSPOT_CHANGE_MACHINE_SLOT:
+                    if self.game_state_manager.game_location == "tr5j":
+                        if self._read_game_state_value_for(12892) == 0:
+                            self._write_game_flags_value_for(12900, 0)
+                        else:
+                            self._write_game_flags_value_for(12900, 2)
+                elif hotspot_item == ZorkGrandInquisitorItems.HOTSPOT_CLOSET_DOOR:
+                    if self.game_state_manager.game_location == "dw1e":
+                        if self._read_game_state_value_for(4983) == 0:
+                            self._write_game_flags_value_for(5010, 0)
+                        else:
+                            self._write_game_flags_value_for(5010, 2)
+                elif hotspot_item == ZorkGrandInquisitorItems.HOTSPOT_CLOSING_THE_TIME_TUNNELS_HAMMER_SLOT:
+                    if self.game_state_manager.game_location == "me2j":
+                        if self._read_game_state_value_for(9491) == 2:
+                            self._write_game_flags_value_for(9539, 0)
+                        else:
+                            self._write_game_flags_value_for(9539, 2)
+                elif hotspot_item == ZorkGrandInquisitorItems.HOTSPOT_CLOSING_THE_TIME_TUNNELS_LEVER:
+                    if self.game_state_manager.game_location == "me2j":
+                        if self._read_game_state_value_for(9546) == 2 or self._read_game_state_value_for(9419) == 1:
+                            self._write_game_flags_value_for(19712, 2)
+                        else:
+                            self._write_game_flags_value_for(19712, 0)
+                elif hotspot_item == ZorkGrandInquisitorItems.HOTSPOT_COOKING_POT:
+                    if self.game_state_manager.game_location == "sg1f":
+                        self._write_game_flags_value_for(2586, 0)
+                elif hotspot_item == ZorkGrandInquisitorItems.HOTSPOT_DENTED_LOCKER:
+                    if self.game_state_manager.game_location == "th3j":
+                        five_is_open: bool = self._read_game_state_value_for(11847) == 1
```

Comment:
can this be anything besides 0 or 1? i could be wrong but i'd think `bool(whatever)` would be faster. if the actual value doesn't matter, python ints are truthy so 0 will eval to false and anything else to true.

Reply:
It can. And I would rather not touch this because it doubles has self-documentation for the precise values I expect to see in RAM

![image](https://github.com/ArchipelagoMW/Archipelago/assets/35039/cd6f1af6-a57b-4a1e-a1a9-2bdf9ce01364). 

I'm not a fan of simplifying without a very good reason to do so. Micro-gains in performance are not worth the obfuscation.

Don't forget that this is technically the mod side of my implementation, not the world side. This is never going to affect world loading, generation etc. Performance is not an issue at all when playing the game.